packages/@sanity/default-layout/src/navbar/Navbar.tsx

```diff
@@ -1,178 +1,328 @@
-                tone="navbar"
+                aria-label="Open menu"
+                icon={MenuIcon}
+                mode="bleed"
+                onClick={onToggleMenu}
+                title="Open menu"
               />
-            </div>
-          </Tooltip>
-        </LegacyLayerProvider>
-      </div>
-      <div className={searchClassName} ref={onSetSearchElement}>
-        <div>
-          <SearchContainer
-            shouldBeFocused={searchIsOpen}
-            onOpen={onSearchOpen}
-            onClose={onSearchClose}
-          />
-        </div>
-      </div>
-      <div className={styles.toolSwitcher}>
-        {tools.length > 1 && (
-          <ToolMenu
-            direction="horizontal"
-            isVisible={showToolMenu}
-            tools={tools}
-            activeToolName={tool}
-            onSwitchTool={onSwitchTool}
-            router={router}
-            showLabel={showLabel}
-            tone="navbar"
-          />
+            </SpacingBox>
+          )}
+
+          {!shouldRender.brandingCenter && (
+            <SpacingBox marginRight={1}>
+              <StateLinkWrap forwardedAs={LinkComponent} state={rootState}>
+                <Branding projectName={config && config.project.name} />
+              </StateLinkWrap>
+            </SpacingBox>
+          )}
+
+          {shouldRender.spaces && (
+            <Flex marginRight={2}>
+              <DatasetSelect />
+            </Flex>
+          )}
+
+          <LegacyLayerProvider zOffset="navbarPopover">
+            <Tooltip
+              portal
+              scheme="light"
+              content={
+                <Box padding={2}>
+                  {createAnyPermission.granted ? (
+                    <Text size={1} muted>
+                      Create new document
+                    </Text>
+                  ) : (
+                    <InsufficientPermissionsMessage
+                      currentUser={currentUser}
+                      operationLabel="create any document"
+                    />
+                  )}
+                </Box>
+              }
+            >
+              <SpacingBox marginRight={shouldRender.brandingCenter ? undefined : 2}>
+                <Button
+                  aria-label="Create new document"
+                  data-testid="default-layout-global-create-button"
+                  icon={ComposeIcon}
+                  mode="bleed"
+                  onClick={onCreateButtonClick}
+                  disabled={!createAnyPermission.granted}
+                  selected={createMenuIsOpen}
+                />
+              </SpacingBox>
+            </Tooltip>
+          </LegacyLayerProvider>
+
+          <LegacyLayerProvider zOffset="navbarPopover">
+            {(searchOpen || !shouldRender.searchFullscreen) && (
+              <SearchCard
+                flex={1}
+                paddingX={shouldRender.searchFullscreen ? 2 : undefined}
+                paddingY={shouldRender.searchFullscreen ? 2 : undefined}
```

Comment:
`paddingX` and `paddingY` here can be replaced by: `padding={shouldRender.searchFullscreen ? 2 : undefined}`